### PR TITLE
Add Team Member Functionality

### PR DESCRIPTION
**Pull request by StephenODea54**
Adds the ability for users to add team members.

image

image

image

### CODE REVIEW

File: src/constants/roles.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+export const ROLES = [
```
Comment (wetherc) [inline src/constants/roles.ts:1]: Do you want me to expose a query to get these dynamically based on edge types?

Comment (StephenODea54) [reply src/constants/roles.ts:1]: Yeah I think that would be beneficial because I would also like to use that when listing the team members.

**Review by wetherc — APPROVED** [review]
Does this conditionally render based on the viewing user's role on the team? Will it only show up as an option for team owners? (Probably glossing over it if you're already doing this --- sorry, on mobile!)

File: src/constants/roles.ts (hunk 1 above)
Comment (wetherc) [reply src/constants/roles.ts:1]: Awesome possum, I'll get on that this evening-ish!

**Comment by wetherc** [comment]
Also how'd you like to handle intersecting all users and current team members? I don't know if we want to have any UI hints since this same mutation can be used to modify existing user roles on a team as well.

We can do it all on the frontend, just intersection of list users and list team members. That'd give you flexibility in how to handle add new member / modify existing / remove existing. Or we could build in some additional logic on the Apollo side of things

**Comment by StephenODea54** [comment]
> Does this conditionally render based on the viewing user's role on the team? Will it only show up as an option for team owners? (Probably glossing over it if you're already doing this --- sorry, on mobile!)

It does not! I was thinking of adding:
- Middleware that redirects page to team page if a member / viewer tries to visit the page.
- Having the "Add Team Member" Button only render if the current user is an owner of that specific team.

**Comment by StephenODea54** [comment]
> Also how'd you like to handle intersecting all users and current team members? I don't know if we want to have any UI hints since this same mutation can be used to modify existing user roles on a team as well.
> 
> We can do it all on the frontend, just intersection of list users and list team members. That'd give you flexibility in how to handle add new member / modify existing / remove existing. Or we could build in some additional logic on the Apollo side of things

Hm, I'm not sure I'm following on the use case for this. Specifically, the mention of "intersecting" users and current team members. Are you referring to situations like only getting the list of users who are not already part of the team they wish to join? Or do you mean extending the current `addToTeam` mutation to also allow for editing a user's role? I think the `addToTeam` already does this because it uses an `UPSERT`, but I just want to double check that with you.

**Pull request merged**